app/decorators/article_decorator.rb

```diff
@@ -102,7 +102,8 @@ def co_authors
   end
 
   def co_author_name_and_path
-    co_authors.map do |user|
+    co_author_ids.map do |id|
+      user = User.select(:name, :username).find(id)
```

Comment:
```suggestion
    User.select(:name, :username).where(id: co_author_ids).in_order_of(:id, co_author_ids).map do |user|
```

This avoids multiple queries instead of one and also avoids an error/crash if the user has been deleted

You can also just change the `co_authors` method to replace `order` with `in_order_of`

Reply:
Thanks @filleduchaos Applying this suggestion.